src/vs/editor/browser/view.ts

```diff
@@ -215,6 +215,7 @@ export class View extends ViewEventHandler {
 
 		if (overflowWidgetsDomNode) {
 			overflowWidgetsDomNode.appendChild(this._contentWidgets.overflowingContentWidgetsDomNode.domNode);
+			overflowWidgetsDomNode.appendChild(this._overlayWidgets.getDomNode().domNode);
```

Comment:
I don't think this is a safe change. We need to group overlay widgets into two buckets, similar to content widgets: some which wish to remain rooted within the editor (the current ones) and some which wish to be overflowing. Otherwise, I think we will break all existing overlay widgets.